src/mongocxx/search_index_view.hpp

```diff
@@ -0,0 +1,246 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <bsoncxx/document/view_or_value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <mongocxx/cursor.hpp>
+#include <mongocxx/search_index_model.hpp>
+
+#include <mongocxx/config/prelude.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Class representing a MongoDB search index view.
+///
+class MONGOCXX_API search_index_view {
+   public:
+    search_index_view(search_index_view&&) noexcept;
```

Comment:
```suggestion
    search_index_view(const search_index_view&) = delete;
    search_index_view& operator=(const search_index_view&) = delete;

    search_index_view(search_index_view&&) noexcept;
```
Rule of 5: explicitly delete the remaining special (copy) member funtions.

Reply:
Alternatively, support copyability as done for `search_index_model` by applying the same `_get_impl() = other._get_impl()` pattern and removing the `= delete` of special copy/move members in `search_index_view::impl`.